mssqlcli/mssql_cli.py

```diff
@@ -167,11 +167,8 @@ def __init__(self, options):
             self.multiline_mode = c['main'].get('multi_line_mode', 'tsql')
             self.vi_mode = c['main'].as_bool('vi')
             self.prompt_format = options.prompt or c['main'].get('prompt', self.default_prompt)
-            if options.row_limit is not None:
-                self.row_limit = options.row_limit
-            else:
-                self.row_limit = c['main'].as_int('row_limit')
-
+            self.row_limit = \
```

Comment:
Is this the right layer/place to make the fix? This seems like an option parsing concern and perhaps handled by argparse?

Reply:
So we definitely can add a check in argparse. However, we may want to keep logic elsewhere because we also accept a value from our config if there's no value for the `--row-limit` arg. Where that *somewhere* should be is open for debate.

I've considered `main.py` and the current location in `mssqlcli.py`. I picked the current location due to easier testability, also it seems like good design to have the constructor handle this logic.

Reply:
Let's fix this to args parse for now - seems like the right place. 

Looks like a there are a number of issues reading values from our config so let's fix those in another PR at a later date.

Reply:
I take back my prior statement—I was able to move logic for checking the config value right in argparse. It seems to work well, once tests pass I'll update the PR.